libraries/node-core-library/src/DiffTest.ts

```diff
@@ -0,0 +1,66 @@
+// Copyright (c) Microsoft Corporation. All rights reserved. Licensed under the MIT license.
+// See LICENSE in the project root for license information.
+
+import * as path from 'path';
+import * as fsx from 'fs-extra';
+
+import { PackageJsonLookup } from './PackageJsonLookup';
+
+/**
+ * Implements a unit testing strategy that generates output files, and then
+ * compares them against the expected input.  If the files are different, then
+ * the test fails.
+ *
+ * @alpha
+ */
+export class DiffTest {
+  private _packageJsonLookup: PackageJsonLookup = new PackageJsonLookup();
+
+  private static _getNormalizedContent(s: string): string {
+    return s.replace(/\r\n/g, '\n').replace(/\r/g, '') // convert to Unix-style newlines
+      .replace(/\s+\n/g, '\n') // strip spaces from end of line
+      .replace(/\n+$/g, '');  // strip newlines from end of file
+  }
+
+  /**
+   * Returns a folder in the temp directory where the unit test should write its output files
+   * to be diffed.
+   * @param unitTestDirName - the "__dirname" variable, evaluated in the context of the unit test
+   * @param testModule - the name of the class being unit tested
+   * @returns A fully qualified path of the folder where the unit test should write its output
+   */
+  public getFolderPath(unitTestDirName: string, testModule: string): string {
```

Comment:
(feedback) Consider writing to a memory buffer and only flushing to disk when the diffs don't match